File: benchmarks/pandas/bench_groupby_min.py
import pandas as pd, time, json
N = 100_000
keys = ["A", "B", "C", "D", "E"]
df = pd.DataFrame({
    "key": [keys[i % len(keys)] for i in range(N)],
    "val": [i * 1.0 for i in range(N)],
})
gb = df.groupby("key")
WARMUP = 3
ITERS = 20
for _ in range(WARMUP):
    gb.min()
t0 = time.perf_counter()
for _ in range(ITERS):
    gb.min()
total = (time.perf_counter() - t0) * 1000
print(json.dumps({"function": "groupby_min", "mean_ms": total / ITERS, "iterations": ITERS, "total_ms": total}))

File: benchmarks/tsb/bench_groupby_min.ts
import { DataFrame, DataFrameGroupBy } from "tsb";
const N = 100_000;
const keys = ["A", "B", "C", "D", "E"];
const df = new DataFrame({
  key: Array.from({ length: N }, (_, i) => keys[i % keys.length]),
  val: Array.from({ length: N }, (_, i) => i * 1.0),
});
const gbObj = new DataFrameGroupBy(df, ["key"]);
const WARMUP = 3;
const ITERS = 20;
for (let i = 0; i < WARMUP; i++) gbObj.min();
const t0 = performance.now();
for (let i = 0; i < ITERS; i++) gbObj.min();
const total = performance.now() - t0;
console.log(JSON.stringify({ function: "groupby_min", mean_ms: total / ITERS, iterations: ITERS, total_ms: total }));
